speedtest.sh

```diff
@@ -7,45 +7,82 @@ DB_HOST="${DB_HOST:-http://localhost:8086}"
 DB_NAME="${DB_NAME:-speedtest}"
 DB_USERNAME="${DB_USERNAME:-admin}"
 DB_PASSWORD="${DB_PASSWORD:-password}"
+SPEEDTEST_HOSTNAME="${SPEEDTEST_HOSTNAME}"
+SPEEDTEST_SERVER_ID="${SPEEDTEST_SERVER_ID}"
 
 run_speedtest()
 {
-    DATE=$(date +%s)
-    HOSTNAME=$(hostname)
+  DATE=$(date +%s)
+  HOSTNAME=$(hostname)
+  CLI_OPTIONS=$1
 
-    # Start speed test
-    echo "Running a Speed Test..."
+  # Start speed test
+  if [ -z "$CLI_OPTIONS" ]; then
+    echo "Running a Speed Test with default host... "
     JSON=$(speedtest --accept-license --accept-gdpr -f json)
-    DOWNLOAD="$(echo $JSON | jq -r '.download.bandwidth')"
-    UPLOAD="$(echo $JSON | jq -r '.upload.bandwidth')"
-    PING="$(echo $JSON | jq -r '.ping.latency')"
-    echo "Your download speed is $(($DOWNLOAD  / 125000 )) Mbps ($DOWNLOAD Bytes/s)."
-    echo "Your upload speed is $(($UPLOAD  / 125000 )) Mbps ($UPLOAD Bytes/s)."
-    echo "Your ping is $PING ms."
-
-    # Save results in the database
-    if $DB_SAVE; 
-    then
-        echo "Saving values to database..."
-        curl -s -S -XPOST "$DB_HOST/write?db=$DB_NAME&precision=s&u=$DB_USERNAME&p=$DB_PASSWORD" \
-            --data-binary "download,host=$HOSTNAME value=$DOWNLOAD $DATE"
-        curl -s -S -XPOST "$DB_HOST/write?db=$DB_NAME&precision=s&u=$DB_USERNAME&p=$DB_PASSWORD" \
-            --data-binary "upload,host=$HOSTNAME value=$UPLOAD $DATE"
-        curl -s -S -XPOST "$DB_HOST/write?db=$DB_NAME&precision=s&u=$DB_USERNAME&p=$DB_PASSWORD" \
-            --data-binary "ping,host=$HOSTNAME value=$PING $DATE"
-        echo "Values saved."
-    fi
+  else
+    echo "Running a Speed Test with options [$CLI_OPTIONS]... "
+    # shellcheck disable=SC2086
+    JSON=$(speedtest --accept-license --accept-gdpr -f json $CLI_OPTIONS)
+  fi
+
+  DOWNLOAD="$(echo $JSON | jq -r '.download.bandwidth')"
+  UPLOAD="$(echo $JSON | jq -r '.upload.bandwidth')"
+  PING="$(echo $JSON | jq -r '.ping.latency')"
+  echo "Your download speed is $(($DOWNLOAD  / 125000 )) Mbps ($DOWNLOAD Bytes/s)."
+  echo "Your upload speed is $(($UPLOAD  / 125000 )) Mbps ($UPLOAD Bytes/s)."
+  echo "Your ping is $PING ms."
+
+  # Save results in the database
+  if $DB_SAVE;
+  then
+      echo "Saving values to database..."
+      curl -s -S -XPOST "$DB_HOST/write?db=$DB_NAME&precision=s&u=$DB_USERNAME&p=$DB_PASSWORD" \
+        --data-binary "download,host=$HOSTNAME value=$DOWNLOAD $DATE"
+      curl -s -S -XPOST "$DB_HOST/write?db=$DB_NAME&precision=s&u=$DB_USERNAME&p=$DB_PASSWORD" \
+        --data-binary "upload,host=$HOSTNAME value=$UPLOAD $DATE"
+      curl -s -S -XPOST "$DB_HOST/write?db=$DB_NAME&precision=s&u=$DB_USERNAME&p=$DB_PASSWORD" \
+        --data-binary "ping,host=$HOSTNAME value=$PING $DATE"
+      echo "Values saved."
+  fi
```

Comment:
There does seem to be a lot of change here, but it is mostly white space, and if you see the diff in my IDE below, you can see I only really added an if statement at #20-#27

- ![git diff in IDE](https://github.com/robinmanuelthiel/speedtest/assets/2152194/aa0b9428-fea2-4557-a3da-8aeb6eeafb6a)